packages/darklang/languageTools/lsp-server/completions.dark

```diff
@@ -6,49 +6,132 @@ module Darklang =
           ()
           : LanguageServerProtocol.Completions.CompletionOptions.CompletionOptions =
           LanguageServerProtocol.Completions.CompletionOptions.CompletionOptions
-            { triggerCharacters = Stdlib.Option.Option.None
-              allCommitCharacters = Stdlib.Option.Option.None
+            { triggerCharacters = Stdlib.Option.Option.Some [ "." ]
+              allCommitCharacters = Stdlib.Option.Option.Some [ "\t" ]
               resolveProvider = Stdlib.Option.Option.Some false // just return all info at once
               completionItem = Stdlib.Option.Option.None }
 
+        let createCompletionItem
+          (label: String)
+          (detail: String)
+          (kind: String)
+          (insertText: String)
+          : LanguageServerProtocol.Completions.CompletionItem.CompletionItem =
+          LanguageServerProtocol.Completions.CompletionItem.CompletionItem
+            { label = label
+              detail = Stdlib.Option.Option.Some detail
+              preselect = Stdlib.Option.Option.Some true
+              sortText = Stdlib.Option.Option.Some label
+              filterText = Stdlib.Option.Option.Some label
+              insertText = Stdlib.Option.Option.Some insertText
+              insertTextFormat =
+                Stdlib.Option.Option.Some
+                  LanguageServerProtocol.Completions.InsertTextFormat.InsertTextFormat.PlainText
+              insertTextMode = Stdlib.Option.Option.None
+              // textEdit = Stdlib.Option.Option.None
+              textEditText = Stdlib.Option.Option.None
+              commitCharacters = Stdlib.Option.Option.Some [ " " ]
+              data = Stdlib.Option.Option.Some(Json.String kind) }
```

Comment:
The `CompletionItem` type has a `kind` field in it, which will affect which icon is shown in the list. If you set up that type, incl the Enum-mapping, you should be able to get "better" (a :package: rather than a :wrench:) icons, rather than simply tossing it in data.